drivers/bluetooth/hci/Kconfig

```diff
@@ -120,6 +120,7 @@ config BT_SPI_INIT_PRIORITY
 
 config BT_BLUENRG_ACI
 	bool "ACI message with with BlueNRG-based devices"
+	select BT_HCI_SET_PUBLIC_ADDR
```

Comment:
I suggest you to remove [this line](https://github.com/arbrauns/zephyr/blob/11fcb16620c30f6c2ec9033213e590d07dc99fd9/drivers/bluetooth/hci/Kconfig#L123) as it forces BlueNRG devices to use public address. Instead, in the definition of `BT_HCI_SET_PUBLIC_ADDR` you can use `depends on BT_BLUENRG_ACI` to make it optional.

Reply:
That doesn't sound right, `BT_HCI_SET_PUBLIC_ADDR` should just not be "forcing to use a public address" in the first place. It should only enable support for setting a public address if desired.

`depends on BT_BLUENRG_ACI` isn't generally correct either, the functionality could be implemented by other drivers as well.

Reply:
Well, I guess other drivers can be added to the depend list and with PR #65988, you are removing the concept of "forcing" which unintentionally, present in this PR.

Reply:
`select` is correct here. It's indicating *support* for a specific feature, not that the feature must be used.